pagerduty/resource_pagerduty_event_orchestration_path_service.go

```diff
@@ -318,14 +319,22 @@ func resourcePagerDutyEventOrchestrationPathServiceUpdate(ctx context.Context, d
 
 func needToUpdateServiceActiveStatus(d *schema.ResourceData) bool {
 	var needToUpdate bool
-	if d.HasChange("enable_event_orchestration_for_service") {
-		o, n := d.GetChange("enable_event_orchestration_for_service")
-		old := o.(bool)
-		new := n.(bool)
-		_, ok := d.GetOkExists("enable_event_orchestration_for_service")
-		if ok || old != new && new == false {
-			needToUpdate = true
-		}
+	o, n := d.GetChange("enable_event_orchestration_for_service")
+	hasChanged := o != n
+	attrVal, isAttrSet := d.GetOkExists("enable_event_orchestration_for_service")
+	isNewResource := d.IsNewResource()
+
+	isNewResAndAttrIsOmitted := !isAttrSet && !attrVal.(bool) && isNewResource
+	isNewResAndAttrIsSetToFalse := isAttrSet && !attrVal.(bool) && isNewResource
+	isNewResAndAttrIsSetToTrue := isAttrSet && attrVal.(bool) && isNewResource
+	isNotNewResAndAttrHasChange := hasChanged && !isNewResource
+
+	if isNewResAndAttrIsOmitted {
+		return false
+	}
+
+	if isNewResAndAttrIsSetToFalse || isNewResAndAttrIsSetToTrue || isNotNewResAndAttrHasChange {
+		needToUpdate = true
```

Comment:
If I'm reading the code correctly, the logic is like this:

1. if it's a new resource AND the attribute is not set -> does NOT need an update (in this case the new service will use EO by default)
2. if it's a new resource and the attribute is set to false -> needs an update
3. if it's a new resource and the attribute is set to true -> needs an update
4. if it's NOT a new resource AND the attribute has changed -> needs an update

Makes sense to me 👍  

**Nit:** I think you can combine options 2 and 3 into a single condition if you want to shave off a line:
```golang
// if it's a new resource and the attribute is set -> needs an update
isNewResAndAttrIsSet := isAttrSet && isNewResource

// ...
if isNewResAndAttrIsSet || isNotNewResAndAttrHasChange {
		needToUpdate = true
}

```